src/main/java/org/primefaces/extensions/component/letteravatar/LetterAvatarRenderer.java

```diff
@@ -43,23 +47,43 @@ public void encodeEnd(final FacesContext context, final UIComponent component) t
         final LetterAvatar letterAvatar = (LetterAvatar) component;
 
         encodeMarkup(context, letterAvatar);
-        encodeScript(context, letterAvatar);
     }
 
     public void encodeMarkup(FacesContext context, LetterAvatar letterAvatar) throws IOException {
         final ResponseWriter writer = context.getResponseWriter();
 
-        final Integer width = letterAvatar.getWidth();
-        final Integer height = letterAvatar.getHeight();
-        final String avatar = letterAvatar.getValue();
+        final String size = letterAvatar.getSize();
+        final String color = letterAvatar.getColor();
+        final String value = letterAvatar.getValue();
+
+        Pattern p = Pattern.compile("\\b[a-zA-Z]");
+        Matcher m = p.matcher(value);
+        StringBuilder sb = new StringBuilder();
+        while (m.find()) {
+            sb.append(m.group());
+        }
+        String initials = sb.toString();
+        // TODO In Java 9, use Matcher#results() to get a Stream<MatchResult>
+//        String initials = Pattern.compile("\\b[a-zA-Z]")
+//                .matcher(value)
+//                .results()
+//                .map(MatchResult::group)
+//                .collect(Collectors.joining(""));
+//              or .toArray(String[]::new);
+        initials = initials.length() == 1 ? initials : initials.charAt(0) + initials.substring(initials.length() - 1);
+
         final Boolean rounded = letterAvatar.isRounded();
 
         final String clientId = letterAvatar.getClientId(context);
-        final String style = letterAvatar.getStyle();
+
+        String backgroundColor = "hsl(" + hue(value) + ", 100%, 50%)";
```

Comment:
If you would allow the user to set the saturation and lightness, that would solve the possible conflict with the text color.